src/test/java/guitests/BoardAutoCreatorTest.java

```diff
@@ -105,14 +105,21 @@ public void sampleBoardAutoCreationTest() {
         waitUntilNodeAppears(SAMPLE_BOARD_DIALOG);
         click("OK");
 
-        assertEquals(panelControl.getPanelCount(), SAMPLE_PANEL_NAMES.size());
+        assertEquals(panelControl.getPanelCount(), BoardAutoCreator.getSamplePanelDetails().size());
         assertEquals(panelControl.getCurrentlySelectedPanel(), Optional.of(0));
         assertEquals(panelControl.getNumberOfSavedBoards(), 1);
+        verifyBoard(panelControl.getCurrentPanelInfos(), BoardAutoCreator.getSamplePanelDetails());
+    }
 
-        List<PanelInfo> panelInfos = panelControl.getCurrentPanelInfos();
-        for (int i = 0; i < SAMPLE_PANEL_NAMES.size(); i++){
-            assertEquals(panelInfos.get(i).getPanelFilter(), SAMPLE_PANEL_FILTERS.get(i));
-            assertEquals(panelInfos.get(i).getPanelName(), SAMPLE_PANEL_NAMES.get(i));
+    /**
+     * Tests whether the panels along with their filters are rightly created and present in the UI
```

Comment:
No it doesn't. The current implementation checks if the panels specified in one parameter is a subset of the other.